peps/pep-0725.rst

```diff
@@ -252,11 +305,40 @@ metadata fields. The ``optional-dependencies`` content from ``[external]``
 would need to either reuse ``Provides-Extra`` or require a new
 ``Provides-External-Extra`` field. Neither seems desirable.
 
+The ``dep:`` URLs MUST be converted to PURL identifiers prior to their
+inclusion in the core metadata by following these rules:
+
+- The ``dep`` *scheme* MUST be replaced with ``pkg``.
+- If the *type* is ``compiler`` or ``interface``, it MUST be transformed
+  into a ``virtual/(compiler|interface)`` type.
+- If present, the *version* field MUST follow these rules:
+
+  - If the value contains no operators, it MUST be kept as is.
+
+  - Otherwise, the version field MUST be removed and its value moved into
+    a ``vers`` qualifier. The *type* of the PURL MUST be prepended to the
+    value, followed by a forward slash. If the *type* does not have a
+    matching ``vers`` type, ``pypi`` MUST be used.
+
+- The PURL percent-encoding rules MUST be applied.
```

Comment:
Those rules are pretty terrible. Wouldn't we be better off with `(ppg, vers)` tuples if it's autogenerated anyway?

Reply:
Turns out that `Requires-External` [does have some syntax rules](https://packaging.python.org/en/latest/specifications/core-metadata/#requires-external-multiple-use) (`name (version); qualifier`), so I rephrased this section so it complies with that loose format.

Reply:
Ah, that explains it, good find!

I think we have to keep it like that then. Or if we want to change it, we must do a core metadata version bump and add new fields for everything.

Reply:
Yes, I kind of like it, even. The only problem is that we can't use any of `();` in PURLs, but I cannot think of examples that require those right now, and if it comes up, %-encoding to the rescue. The packaging docs do not cover that case either, so let's leave it to parsers to figure it our (split on ; from the right, regex on () on the left side, etc).